sdks/python/apache_beam/pipeline.py

```diff
@@ -191,47 +189,73 @@ class TransformUpdater(PipelineVisitor): # pylint: disable=used-before-assignmen
       def __init__(self, pipeline):
         self.pipeline = pipeline
 
-      def _replace_if_needed(self, transform_node):
-        if matcher(transform_node):
+      def _replace_if_needed(self, original_transform_node):
+        if matcher(original_transform_node):
+          assert isinstance(original_transform_node, AppliedPTransform)
           replacement_transform = override.get_replacement_transform(
-              transform_node.transform)
-          inputs = transform_node.inputs
-          # TODO:  Support replacing PTransforms with multiple inputs.
+              original_transform_node.transform)
+
+          replacement_trasform_node = AppliedPTransform(
+              original_transform_node.parent, replacement_transform,
+              original_transform_node.full_label,
+              original_transform_node.inputs)
+
+          # Transform execution could depend on order in which nodes are
+          # considered. Hence we insert the replacement transform node to same
+          # index as the original transform node. Note that this operation
+          # removes the original transform node.
+          if original_transform_node.parent:
+            assert isinstance(original_transform_node.parent, AppliedPTransform)
+            parent_parts = original_transform_node.parent.parts
+            parent_parts[parent_parts.index(original_transform_node)] = (
+                replacement_trasform_node)
+          else:
+            # Original transform has to be a root
```

Comment:
<!--new_thread; commit:eb3d1308494b34cb40336a41f40937e9b74c266a; resolved:0-->
Period at end of sentence.